common/db/workspace.go

```diff
@@ -0,0 +1,200 @@
+package db
+
+import (
+	"GADS/common/models"
+	"context"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+func AddWorkspace(workspace *models.Workspace) error {
```

Comment:
If we are not going to keep those objects somewhere in memory that has to be updated we don't really to use a pointer, it doesn't really matter though as long as it works

Reply:
In this [commit](https://github.com/gifflet/GADS/commit/68c3e94a31bc8cb37c9de49884f9fbde25605bdb), we are using a pointer to the workspace because we apply the newly generated workspace ID from MongoDB. I observed that after creating a workspace and trying to use it to associate users or workspaces, its ID value is an empty string ("") since MongoDB doesn’t assign an ID as fast as we try to use its value.

```go
func AddWorkspace(workspace *models.Workspace) error {
 	collection := mongoClient.Database("gads").Collection("workspaces")
 	result, err := collection.InsertOne(mongoClientCtx, workspace)
 	if err != nil {
 		return err
 	}
 	workspace.ID = result.InsertedID.(primitive.ObjectID).Hex()
 	return nil
 }
```

Reply:
Weird, the MongoDB driver [docs](https://www.mongodb.com/docs/manual/reference/method/db.collection.insertOne/) claim it should be returned... Maybe we should update the package hoping it won't break stuff :D

Reply:
The docs state that the ID value is returned through the `InsertedID` field, so the current implementation is correct, right?

Reply:
yes, I believe so, I can see that the ID value is returned in the result currently 👌